### PR TITLE
Source debug documents

### PR DESCRIPTION
**Pull request by xujuntwt95329**

### CODE REVIEW

File: doc/source_debugging.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# WAMR source debugging
+
+WAMR supports source level debugging based on DWARF, source map is not supported.
```
Comment (wenyongh) [inline doc/source_debugging.md:3]: Should we add more details for DWARF and source map, developer may be confused, e.g.:
WAMR supports source level debugging based on DWARF (normally used in C/C++/Rust), source map (normally used in JavaScript/TypeScript) is not supported.

File: doc/source_debugging.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+Then you will get `test.wasm` which is a WebAssembly module with embedded DWARF sections. Further, you can use `llvm-dwarfdump` to check if the generated wasm file contains DWARF information:
+``` bash
+llvm-dwarfdump-12 test.wasm
+```
+
+## Debugging with interpreter
+
+1. Build iwasm with source debugging feature
```
Comment (wenyongh) [inline doc/source_debugging.md:18]: Add a step of how to build lldb with patch?

File: doc/source_debugging.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+lldb
+(lldb) gdb-remote 127.0.0.1:1234
+```
+Then you can use lldb commands to debug your applications. Please refer to [lldb document](https://lldb.llvm.org/use/tutorial.html) for command usage.
+
+## Debugging with AoT
+
+> Note: AoT debugging is experimental and only a few debugging capabilities are supported.
+
+1. Build lldb (assume you have already built llvm)
```
Comment (wenyongh) [inline doc/source_debugging.md:42]: 是不是apt-get lldb也可以？只要lldb版本大于12。如果是的话，最好加一下说明，编译一遍可能比较麻烦？

File: README.md [hunk 4]
```diff
@@ -28,6 +28,7 @@ iwasm VM core
 - [Multiple modules as dependencies](./doc/multi_module.md), ref to [sample](samples/multi-module)
 - [Thread management and pthread library](./doc/pthread_library.md), ref to [sample](samples/multi-thread)
 - [Linux SGX (Intel Software Guard Extension) support](./doc/linux_sgx.md)
+- [Source debugging support](./doc/source_debugging.md)
```
Comment (wenyongh) [inline README.md:31]: 感觉去掉support好点？上面SGX不写support总感觉不大完整，所以加了

File: doc/source_debugging.md (hunk 1 above)
Comment (xujuntwt95329) [reply doc/source_debugging.md]: Done

File: doc/source_debugging.md (hunk 2 above)
Comment (xujuntwt95329) [reply doc/source_debugging.md:18]: Done

File: doc/source_debugging.md (hunk 3 above)
Comment (xujuntwt95329) [reply doc/source_debugging.md:51]: If we want to use lldb installed by apt, then we'll also need to use llvm installed by apt

File: README.md (hunk 4 above)
Comment (xujuntwt95329) [reply README.md]: Done

**Pull request merged**